src/phy/mod.rs

```diff
@@ -239,7 +239,7 @@ pub trait Device {
 }
 
 pub trait RxToken {
-    fn consume<R, F: FnOnce(&[u8]) -> R>(self, timestamp: u64, f: F) -> R;
+    fn consume<R, F: FnOnce(&[u8]) -> Result<R>>(self, timestamp: u64, f: F) -> Result<R>;
```

Comment:
Because we want to return an `Err` from some middleware `consume` functions. One example is tap_interface:

```rust
fn consume<R, F: FnOnce(&[u8]) -> Result<R>>(self, _timestamp: u64, f: F) -> Result<R> {
    […]
   match lower.recv(&mut buffer[..]) {
       Ok(size) => […]
       Err(ref err) if err.kind() == io::ErrorKind::WouldBlock => {
            Err(Error::Exhausted)
        }
        Err(err) => panic!("{}", err)
    }
}
```

If we returned `R`, we would need to call `f` with a dummy packet in case of error to get a valid return type.

Reply:
Shouldn't we also give `TxToken::consume` similar treatment, for the same reason?

Reply:
I implemented it